tests/virt_autotest/reboot_and_wait_up_upgrade.pm

```diff
@@ -48,4 +48,18 @@ sub run {
     $self->reboot_and_wait_up($timeout);
 }
 
+sub enable_y2log_analysis {
+    # Create and Upload y2log for analysis
+    assert_script_run "save_y2logs /tmp/y2logs.tar.bz2", 180;
+    upload_logs("/tmp/y2logs.tar.bz2");
+    save_screenshot;
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+    reset_consoles;
+    select_console('root-console');
+    $self->enable_y2log_analysis if (check_var('offline_upgrade', 'yes'));
```

Comment:
Shall it also call baseclass's post_fail_hook?

Reply:
yes, it also call aseclass's post_fail_hook. 
had added aseclass's post_fail_hook.

Reply:
DONE